modules/ROOT/pages/sync/authorization.adoc

```diff
@@ -0,0 +1,110 @@
+include::{partialsdir}/attributes.adoc[]
+
+:service-name: Data Sync
+
+= Role Based Authorization in {sync-service}
+
+== Prerequisites
+
+* xref:./binding_to_keycloak.adoc[Binding {sync-service} to Keycloak].
+* link:https://www.keycloak.org/docs/3.4/getting_started/index.html#creating-a-realm-and-user[Creating a Realm in Keycloak].
+* link:https://www.keycloak.org/docs/3.4/authorization_services/index.html#_getting_started_hello_world_enabling_authz_services[Enabling Authorization Services in Keycloak].
+* link:https://www.keycloak.org/docs/3.4/authorization_services/index.html#_policy_overview[Managing Policies in Keycloak]
+
+
+Role based authorization can be applied to Queries, Mutations, Subscriptions and individual fields in a GraphQL schema, when {sync-service} is used with Keycloak. This means it is possible to restrict individual fields and/or operations defined in the GraphQL schema to users with specific roles in Keycloak.
+
+Take a publishing platform as an example. Role Based Authorization could be used to ensure only users with the role `editor` can perform the `editPublication` mutation.
+
+{sync-service} currently supports authorization using **client** roles and **realm** roles defined in keycloak. Consult the Keycloak documentation to learn how to create roles.
+
+* link:https://www.keycloak.org/docs/4.3/server_admin/index.html#realm-roles[Realm Roles in Keycloak]
+* link:https://www.keycloak.org/docs/4.3/server_admin/index.html#client-roles[Client Roles in Keycloak]
+
+== The @hasRole Directive
+
+Role based authorization can be defined using the `@hasRole` directive. The `@hasRole` directive is a special annotation that can be applied to
+
+* Fields
+* Queries
+* Mutations
+* Subscriptions
+
+The `@hasRole` usage is as follows:
+
+* `@hasRole(role: String)`
+  * Example - `@hasRole(role: "admin])`
+  * If the authenticated user has the role `admin` they will be authorized.
+* `@hasRole(role: [String])`
+  * Example - `@hasRole(role: ["admin", "editor"])`
+  * If the authenticated user has **at least one of the roles** in the list, they will be authorized.
+
+**The default behaviour is to check client roles.** For example, `@hasRole(role: "admin")` will check that a user has a client role called `admin`.
```

Comment:
This section is no longer actual.

Reply:
@wtrocki it is indeed still true. `hasRole(role: "admin") well check against client roles, whereas `hasRole(role: "realm:admin") will check against a realm role.

Reply:
ah.. right. Yes. That is true